result/result.py

```diff
@@ -1,211 +1,245 @@
+    def __repr__(self) -> str:
+        return "Ok({})".format(repr(self._value))
 
     def __eq__(self, other: Any) -> bool:
-        return (self.__class__ == other.__class__ and
-                self.is_ok() == cast(Result, other).is_ok() and
-                self._value == other._value)
+        return isinstance(other, Ok) and self.value == other.value
 
     def __ne__(self, other: Any) -> bool:
         return not (self == other)
 
     def __hash__(self) -> int:
-        return hash((self.is_ok(), self._value))
+        return hash((True, self._value))
 
-    def __repr__(self) -> str:
-        if self.is_ok():
-            return 'Ok({})'.format(repr(self._value))
-        else:
-            return 'Err({})'.format(repr(self._value))
+    def is_ok(self) -> bool:
+        return True
 
-    @classmethod
-    @overload
-    def Ok(cls) -> 'Result[bool, Any]':
-        pass
+    def is_err(self) -> bool:
+        return False
 
-    @classmethod
-    @overload
-    def Ok(cls, value: T) -> 'Result[T, Any]':
-        pass
+    def ok(self) -> T:
+        """
+        Return the value.
+        """
+        return self._value
+
+    def err(self) -> None:
+        """
+        Return `None`.
+        """
+        return None
 
-    @classmethod
-    def Ok(cls, value: Any = True) -> 'Result[Any, Any]':
-        return cls(is_ok=True, value=value, force=True)
+    @property
+    def value(self) -> T:
+        """
+        Return the inner value.
+        """
+        return self._value
+
+    def expect(self, _message: str) -> T:
+        """
+        Return the value.
+        """
+        return self._value
 
-    @classmethod
-    def Err(cls, error: E) -> 'Result[Any, E]':
-        return cls(is_ok=False, value=error, force=True)
+    def expect_err(self, _message: str) -> T:
+        """
+        Raise an UnwrapError since this type is `Ok`
+        """
+        raise UnwrapError(_message)
+
+    def unwrap(self) -> T:
+        """
+        Return the value.
+        """
+        return self._value
+
+    def unwrap_err(self) -> NoReturn:
+        """
+        Raise an UnwrapError since this type is `Ok`
+        """
+        raise UnwrapError("Called `Result.unwrap_err()` on an `Ok` value")
+
+    def unwrap_or(self, _default: T) -> T:
+        """
+        Return the value.
+        """
+        return self._value
+
+    def map(self, op: Callable[[T], U]) -> 'Result[U, E]':
+        """
+        The contained result is `Ok`, so return `Ok` with original value mapped to
+        a new value using the passed in function.
+        """
+        return Ok(op(self._value))
+
+    def map_or(self, default: U, op: Callable[[T], U]) -> U:
+        """
+        The contained result is `Ok`, so return the original value mapped to a new
+        value using the passed in function.
+        """
+        return op(self._value)
+
+    def map_or_else(
+            self,
+            default_op: Callable[[], U],
+            op: Callable[[T], U]
```

Comment:
Personal preference: Can you indent these values only 1 level (4 spaces)?

```suggestion
        self,
        default_op: Callable[[], U],
        op: Callable[[T], U]
```

Reply:
Fixed in 947814d4a63cb4e34766cd7eb2538e0ab527c1b8